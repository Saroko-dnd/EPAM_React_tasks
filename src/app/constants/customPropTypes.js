import { createPropType } from 'react-custom-proptypes';

const articlePropType = createPropType(prop => (
    typeof prop === 'object' &&
        typeof prop.author === 'string' &&
        typeof prop.title === 'string' &&
        typeof prop.description === 'string' &&
        typeof prop.url === 'string' &&
        typeof prop.urlToImage === 'string' &&
        typeof prop.publishedAt === 'string' &&
        typeof prop.source === 'object' &&
        typeof prop.source.name === 'string'
), 'Must be news article');

export default { article: articlePropType };
