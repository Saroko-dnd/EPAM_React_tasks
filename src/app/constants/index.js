const newsAppConfig = {
    apiToken: 'e99a027c252b468ab4c0d3d8872f4cfc',
    apiBaseLink: 'https://newsapi.org/v2/',

    navigation: {
        homePage: { destination: '/', title: 'Top news', id: 'nav-link-1' },
        firstPage: {
            destination: '/first',
            title: 'Ultra news',
            id: 'nav-link-2',
        },
        secondPage: {
            destination: '/second',
            title: 'Super news',
            id: 'nav-link-3',
        },
        thirdPage: {
            destination: '/third',
            title: 'Mega news',
            id: 'nav-link-4',
        },
    },

    colors: {
        green: '#009900',
    },
};

export default newsAppConfig;
