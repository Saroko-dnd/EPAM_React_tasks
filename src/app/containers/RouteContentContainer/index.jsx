import React from 'react';
import PropTypes from 'prop-types';
import { Switch, Route } from 'react-router-dom';

import NewsColumns from '../NewsColumns';
import SelectedArticle from '../SelectedArticle';
import { api } from '../../constants';

const RouteContentContainer = ({ className }) => (
    <main>
        <Switch>
            <Route
                exact
                path="/"
                render={() => (
                    <div className={`${className} container-fluid`}>
                        <NewsColumns
                            newsApiLink={api.apiTopNewsLink}
                            newsApiToken={api.apiToken}
                        />
                    </div>
                )}
            />
            <Route exact path="/details" component={SelectedArticle} />
            <Route exact path="/second" render={() => <h1>Route 3</h1>} />
            <Route exact path="/third" render={() => <h1>Route 4</h1>} />
        </Switch>
    </main>
);

RouteContentContainer.propTypes = {
    className: PropTypes.string,
};

RouteContentContainer.defaultProps = {
    className: '',
};

export default RouteContentContainer;
