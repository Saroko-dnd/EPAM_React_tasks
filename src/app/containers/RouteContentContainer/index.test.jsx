import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import RouteContentContainer from './index';
import { initialState } from '../../constants';

describe('RouteContentContainer component (routing test)', () => {
    describe('rendering and behavior', () => {
        const mockStore = configureStore();
        const store = mockStore(initialState);

        it('should render top news columns if path = "/"', () => {
            const router = (
                <Provider store={store}>
                    <MemoryRouter initialEntries={['/']}>
                        <RouteContentContainer />
                    </MemoryRouter>
                </Provider>
            );
            const routerWrapper = render(router);

            expect(routerWrapper).toMatchSnapshot();
        });

        it('should render slider with detailed news and news related to the selected news if path = "/details"', () => {
            const router = (
                <Provider store={store}>
                    <MemoryRouter initialEntries={['/details']}>
                        <RouteContentContainer />
                    </MemoryRouter>
                </Provider>
            );
            const routerWrapper = render(router);

            expect(routerWrapper).toMatchSnapshot();
        });

        it('should render "Route 3" message if path = "/second"', () => {
            const router = (
                <Provider store={store}>
                    <MemoryRouter initialEntries={['/second']}>
                        <RouteContentContainer />
                    </MemoryRouter>
                </Provider>
            );
            const routerWrapper = render(router);

            expect(routerWrapper).toMatchSnapshot();
        });

        it('should render "Route 4" message if path = "/third"', () => {
            const router = (
                <Provider store={store}>
                    <MemoryRouter initialEntries={['/third']}>
                        <RouteContentContainer />
                    </MemoryRouter>
                </Provider>
            );
            const routerWrapper = render(router);

            expect(routerWrapper).toMatchSnapshot();
        });
    });
});
