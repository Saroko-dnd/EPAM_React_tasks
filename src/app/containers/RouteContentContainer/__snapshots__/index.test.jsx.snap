// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteContentContainer component (routing test) rendering and behavior should render "Route 3" message if path = "/second" 1`] = `
<main>
  <h1>
    Route 3
  </h1>
</main>
`;

exports[`RouteContentContainer component (routing test) rendering and behavior should render "Route 4" message if path = "/third" 1`] = `
<main>
  <h1>
    Route 4
  </h1>
</main>
`;

exports[`RouteContentContainer component (routing test) rendering and behavior should render slider with detailed news and news related to the selected news if path = "/details" 1`] = `
<main>
  <section
    class="d-flex align-items-center justify-content-center flex-column"
  >
    <div
      class="carousel-wrapper"
    />
    <div
      class="news-list "
    >
      <h2
        class="title"
      >
        Related news
      </h2>
      <div
        class="d-flex justify-content-center align-items-center flex-wrap justify-content-center"
      />
    </div>
  </section>
</main>
`;

exports[`RouteContentContainer component (routing test) rendering and behavior should render top news columns if path = "/" 1`] = `
<main>
  <div
    class=" container-fluid"
  >
    <section
      class="news-columns row justify-content-center align-items-stretch"
    />
  </div>
</main>
`;
